Source/WebCore/html/CanvasBase.cpp

```diff
@@ -356,6 +358,88 @@ RefPtr<ImageBuffer> CanvasBase::allocateImageBuffer(bool usesDisplayListDrawing,
     return ImageBuffer::create(size(), RenderingPurpose::Canvas, 1, colorSpace, pixelFormat, bufferOptions, context);
 }
 
+bool CanvasBase::postProcessPixelBuffer(Ref<PixelBuffer> pixelBuffer, bool wasLastDrawByBitMap, const HashSet<uint32_t>& suppliedColors) const
+{
+    if (!scriptExecutionContext() || !scriptExecutionContext()->noiseInjectionHashSalt() || wasLastDrawByBitMap || !scriptExecutionContext()->settingsValues().canvasNoiseInjectionEnabled)
+        return false;
+
+    constexpr auto contextString { "Canvas2DContextString"_s };
+    unsigned salt = computeHash<String, uint64_t>(contextString, *scriptExecutionContext()->noiseInjectionHashSalt());
+    constexpr int bytesPerPixel = 4;
+    auto* bytes = pixelBuffer->bytes();
+    HashMap<uint32_t, uint32_t> pixelColorMap;
+    bool wasPixelBufferModified { false };
+
+    for (size_t i = 0; i < pixelBuffer->sizeInBytes(); i += bytesPerPixel) {
+        auto& redChannel = bytes[i];
+        auto& greenChannel = bytes[i+1];
+        auto& blueChannel = bytes[i+2];
+        auto& alphaChannel = bytes[i+3];
```

Comment:
Is the format of this `PixelBuffer` always going to be RGBA? (If so, we could perhaps add an `ASSERT(pixelBuffer->format().pixelFormat == PixelFormat::RGBA8);`)

Reply:
I believe it will be, and we'll end up with garbage data if it's not but we handle it as if it is below...so we should add the assert.